### ISSUE TITLE
2.0 create `provide` that takes object

### ISSUE THREAD

**Issue by domdorn** [zio/zio#6661]
```scala
  private val controllers: ZLayer[AcmeBuildInfos, Nothing, BuildinfosController] = BuildinfosController.live
  
  def liveEffect(buildInfos: AcmeBuildInfos): ZIO[Any, Nothing, WellKnownModule] = {
    for {
      controllers <- ZIO.service[BuildinfosController]
    } yield WellKnownModule(controllers)
  }.provide(ZLayer.succeed(buildInfos))
  ```
  
  it would be nice if there's a way to have a `provide` where I don't have to wrap the object in a Layer first

**Comment by adamgfraser** [comment]
This seems like a bit too much magic to me. `provide` is an operator that takes a collection of layers and provides them to the workflow. Having it take a collection of things of arbitrary types some of which are layers and some of which are not layers seems quite confusing.

**Comment by adamgfraser** [comment]
Note that there is also already `provideService` for providing a single service that is not a layer.

**Comment by domdorn** [comment]
hmm.. I was expecting `provde` to be smart enough for layers and other types.. maybe it would be better to get rid of `provide` and call it `provideLayer` or `provideLayers` instead.. then people will type `.provide<tab>` and see "ah ok, these methods are for layers, these are for services".. ?

**Comment by adamgfraser** [comment]
We're really trying to focus people on the simplest thing. Layers are the idiomatic way of constructing services.

**Comment by domdorn** [comment]
>> We're really trying to focus people on the simplest thing. Layers are the idiomatic way of constructing services.
yeah, I get that.. at the moment its "ok, this effect needs x, I'll just `provide` that" and then people are wondering why they get a cryptic compile error. If they would be offered exactly two methods - one for layers, one for instances - it would be clear what to choose..

**Comment by adamgfraser** [comment]
If anything I would just delete the variant that accepts something other than a layer. The thing you provide to your workflow is a layer. If you have something else you turn it into a layer. We used to have a lot more variants of provide that took ZIO workflows, managed values, and so on and we worked very hard to simplify that. The variant that accepts a single service was a convenience method for a very particular case of implementing service operators but maybe that was a mistake, especially with having removed the default services from the environment.
